venues/NeurIPS.cc/2023/submission_preprocess.js

```diff
@@ -0,0 +1,73 @@
+async function process(client, edit, invitation) {
+    client.throwErrors = true
+    
+    const { note } = edit
+
+    if (note.ddate) { 
+      return
+    }
+
+    const profiles = await client.tools.getProfiles(note.content.authorids.value)
+    const profileIds = profiles.map(profile => profile.id)
+
+    const primaryAuthorField = note.content.corresponding_author.value
+    const primaryAuthorProfile = await client.tools.getProfiles([primaryAuthorField])
+    const primaryAuthor = primaryAuthorProfile[0].id
+    if (!profileIds.includes(primaryAuthor)) {
+      return Promise.reject(new OpenReviewError({ name: 'Error', message: 'Select a paper co-author as corresponding author, ' + primaryAuthorField + ' does not appear in the author list' }))
+    }
+
+    if (note.content.financial_support) {
+      const financialSupportField = note.content.financial_support.value
+      const financialSupportProfile = await client.tools.getProfiles([financialSupportField])
+      const financialSupport = financialSupportProfile[0].id
+      if (!profileIds.includes(financialSupport)) {
+        return Promise.reject(new OpenReviewError({ name: 'Error', message: 'Select a paper co-author for financial support, ' + financialSupportField + ' does not appear in the author list' }))
+      }
+    }
+
+    const reviewerNominationField = note.content.reviewer_nomination.value
+    const reviewerNominationProfile = await client.tools.getProfiles([reviewerNominationField])
+    const reviewerNomination = reviewerNominationProfile[0].id
+    if (!profileIds.includes(reviewerNomination)) {
+      return Promise.reject(new OpenReviewError({ name: 'Error', message: 'Select a paper co-author to nominate as reviewer, ' + reviewerNominationField + ' does not appear in the author list' }))
+    }
+
+    const { groups: revGroups } = await client.getGroups({ id: invitation.domain + '/Reviewers' })
+    const reviewers = revGroups[0].members;
+    
+    const reviewerMembers = {}
+    for (const reviewer of reviewers) {
+      reviewerMembers[reviewer] = true
+    }
+
+    for (const profile of profiles) {
+      const emails = profile.content.emails
+      const usernames = profile.content.names.map(name => name.username).filter(username => username);
+      const cleanUsernames = usernames.filter(function( element ) {
```

Comment:
I guess this filter function can be in the above line, right?

```
const usernames = profile.content.names.map(name => name.username).filter(username => username !== undefined);
```

cc @carlosmondra

Reply:
ohh my bad, Carlos gave me the suggestion of doing it in one line and I forgot to remove my implementation to get rid of undefined